presto-jdbc/src/main/java/com/facebook/presto/jdbc/PrestoConnectionConfig.java

```diff
@@ -39,26 +49,33 @@
 
     private final HostAndPort address;
     private final URI uri;
+    private final Properties connectionProperties;
 
     private String catalog;
     private String schema;
 
     private final boolean useSecureConnection;
 
-    public PrestoDriverUri(String url)
+    private static final Map<ConnectionProperty, BiConsumer<HttpClientConfig, String>> CLIENT_SETTERS = ImmutableMap.of(
```

Comment:
It lets us express the relationship between ConnectionProperties and HttpClient config through a mapping, which is what it is conceptually.